spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/config/LocalResponseCacheAutoConfiguration.java

```diff
@@ -41,12 +41,12 @@
 
 /**
  * @author Ignacio Lozano
+ * @author Marta Medio
  */
 @Configuration(proxyBeanMethods = false)
 @EnableConfigurationProperties({ LocalResponseCacheProperties.class })
 @ConditionalOnClass({ Weigher.class, Caffeine.class, CaffeineCacheManager.class })
-@ConditionalOnProperty(name = "spring.cloud.gateway.enabled", matchIfMissing = true)
-@ConditionalOnEnabledFilter(LocalResponseCacheGatewayFilterFactory.class)
+@ConditionalOnExpression("${spring.cloud.gateway.enabled:true} && ${spring.cloud.gateway.filter.local-response-cache.enabled:false}")
```

Comment:
Change this to a custom `Conditional` see
https://github.com/spring-cloud/spring-cloud-netflix/blob/e6c8caeb0f5d616ee96cfe850a81434e11e411cd/spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientAutoConfiguration.java#L366-L387

and https://github.com/spring-cloud/spring-cloud-netflix/blob/e6c8caeb0f5d616ee96cfe850a81434e11e411cd/spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientAutoConfiguration.java#L351

Reply:
Nice!
Done using a custom `Conditional`: https://github.com/spring-cloud/spring-cloud-gateway/pull/2848/commits/9be5e7eb4b50997cd1da035f92489f7be1ac4c02